subset_vds.py

```diff
@@ -335,14 +340,14 @@ def main(
         else:
             subset_vds = subset_by_variants(input_variants, keep_variants, input_vds)
 
-    if n_samples:
-        subset_sample_list = get_subset_sample_list(input_vds, n_samples)
+    if n_samples or sample_list:
+        subset_sample_list = sample_list or get_subset_sample_list(input_vds, n_samples)
```

Comment:
What happens here if someone provides both a sample list and `n_samples`? I suggest that we add a check earlier to only allow one.